src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -1445,10 +1446,36 @@ int RGWBucketAdminOp::remove_bucket(rgw::sal::Driver* driver, RGWBucketAdminOpSt
   if (ret < 0)
     return ret;
 
+  // check if the bucket is owned by my zonegroup, if not, refuse to remove it
+  if (!forwarded_request && bucket->get_info().zonegroup != driver->get_zone()->get_zonegroup().get_id()) {
+    ldpp_dout(dpp, 0) << "ERROR: The bucket's zonegroup does not match. "
+                      << "Removal is not allowed. Please execute this operation "
+                      << "on the bucket's zonegroup: " << bucket->get_info().zonegroup << dendl;
+    return -ERR_PERMANENT_REDIRECT;
+  }
+
   if (bypass_gc)
     ret = bucket->remove_bypass_gc(op_state.get_max_aio(), keep_index_consistent, y, dpp);
   else
     ret = bucket->remove(dpp, op_state.will_delete_children(), y);
+  if (ret < 0)
+    return ret;
+
+  // forward to master zonegroup
+  const std::string delpath = "/" + dpp->get_cct()->_conf->rgw_admin_entry + "/bucket";
```

Comment:
concerned about the use of `rgw_admin_entry` here, because it only applies to the local radosgw instance

rgw_data_sync.cc uses hard-coded paths like `/admin/log`, so multisite wouldn't work at all if the remote changed `rgw_admin_entry`

consider hard-coding it here too, and maybe adding a note to `rgw_admin_entry` that it shouldn't be changed in multisite configurations

Reply:
agree, We might need to introduce a `realm configuration` that can be synced over a multisite realm to make it consistent.